### PR TITLE
grc-qt: grc --qt filename.grc doesn't open filename.grc

### LINKED ISSUE #7291: gnuradio-companion --qt filename.grc doesn't open filename.grc
### What happened?

Tried to open a GRC flowgraph directly with QT GRC; it doesn't do that, doesn't throw an error. The same file works when opened via menu.

Doesn't work without the --qt if the default GRC variant is set to QT in the preferences, either.

### System Information

OS: linux
GR Installation Method: source

### GNU Radio Version

3.11-git (main)

### Specific Version

v3.11.0.0git-716-g77155087

### Steps to Reproduce the Problem

gnuradio-companion --qt filename.grc

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by dl1ksv**
## Description

Now the list of files specified on the cl for grc is appended to the list of files to be opened on startup.

fixes #7291

## Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md).
- [x] I have squashed my commits to have one significant change per commit. 
- [x] I [have signed my commits before making this PR](https://github.com/gnuradio/gnuradio/blob/main/CONTRIBUTING.md#dco-signed)
- [ ] My code follows the code style of this project. See [GREP1.md](https://github.com/gnuradio/greps/blob/main/grep-0001-coding-guidelines.md).
- [ ] I have updated [the documentation](https://wiki.gnuradio.org/index.php/Main_Page#Documentation) where necessary.
- [ ] I have added tests to cover my changes, and all previous tests pass.

### CODE REVIEW

**Review by marcusmueller — APPROVED** [review]
Looks good, tested it and it works!

File: grc/gui_qt/components/window.py [hunk 1]
```diff
@@ -66,7 +66,7 @@
 
 
 class MainWindow(QtWidgets.QMainWindow, base.Component):
-    def __init__(self):
+    def __init__(self, file_path=[]):
```
Comment (marcusmueller) [inline grc/gui_qt/components/window.py:69]: Setting a default `= []` argument *can* be tricky in situations where the function modifies the parameter; the default parameter is only constructed once and then it stays the same object forever. For example:

```python
def print_list(printme = []):
  printme += ["end"]
  print(printme)

print("first call:")
print_list()

print("second call:")
print_list()
```

would print
```text
first call:
['end']
second call:
['end', 'end']
```

Comment (marcusmueller) [reply grc/gui_qt/components/window.py:69]: The typical workaround is to use a `None` default parameter:
```python
def better_print_list(printme = None):
  if not printme:
   printme = []
  printme += ["end"]
  print(printme)
```

**Review by marcusmueller — COMMENTED** [review]
Not a criticism, because the usage here is IMHO absolutely OK (`file_path` would not reasonably ever be changed in `__init__`); really just a remark:

File: grc/gui_qt/components/window.py (hunk 1 above)
Comment (dl1ksv) [reply grc/gui_qt/components/window.py:69]: Thanks for your clarification. I used [] as default, as this is just the value you get from the cli if you enter no grc file.  So I can just add file_path to files_open. And as you mentioned file_path will not be changed in __init__

**Pull request merged**